src/encounterState/index.js

```diff
@@ -0,0 +1,70 @@
+import React, {
+  useReducer,
+  createContext,
+  useContext,
+  useCallback,
+} from "react";
+
+import { reducer, EncounterActions } from "./reducer";
+
+const initialContext = {
+  isLoading: false,
+  npc: null,
+};
+
+const StateContext = createContext(initialContext);
+const DispatchContext = createContext(undefined);
+
+export const EncounterContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducer, initialContext);
+
+  return (
+    <StateContext.Provider value={state}>
+      <DispatchContext.Provider value={dispatch}>
+        {children}
+      </DispatchContext.Provider>
+    </StateContext.Provider>
+  );
+};
+
+export const useEncounterState = () => useContext(StateContext);
+
+export const useEncounterDispatch = () => {
+  const dispatch = useContext(DispatchContext);
+
+  if (dispatch === undefined) {
+    throw new Error("useDispatch used outside of provider");
+  }
+
+  const getNewEncounter = useCallback(
+    async (area) => {
+      dispatch({ type: EncounterActions.NEW_ENCOUNTER_LOAD });
+
+      const allNpcs = await (
+        await fetch("http://localhost:8000/api/v1/npcs")
```

Comment:
We probably want to change this to be an environment variable so when we deploy on production we can change the url

Reply:
Of course, this was just to get things running. Will look into environment variables shortly